artichoke-backend/src/extn/core/env/env.rb

```diff
@@ -36,10 +36,58 @@ def delete_if
     to_h
   end
 
+  def each
+    return to_enum(:each) unless block_given?
+
+    to_h.each { |key, value| yield key, value }
+  end
+
+  def each_key
+    return to_enum(:each_key) unless block_given?
+
+    to_h.each_key { |key| yield key }
+  end
+
+  def each_pair
+    return to_enum(:each) unless block_given?
+
+    to_h.each_pair { |key, value| yield key, value }
+  end
+
+  def each_value
+    return to_enum(:each_value) unless block_given?
+
+    to_h.each_value { |value| yield value }
+  end
+
   def empty?
     to_h.empty?
   end
 
+  def fetch(name, default = nil)
+    value = self[name]
+
+    return value unless value.nil?
+
+    return yield name if block_given?
+
+    return default unless default.nil?
```

Comment:
ENV.fetch with an explicit nil default returns nil for a not found key.

One way we can implement this is with an expression as the default value:

```ruby
def fetch(name, default = (not_set = true))
  # ...
end
```

If no default is supplied, the expression is evaluated. Then we can `return default if not_set.nil?`

Reply:
Marvelous. Ruby is so tricky :-)

Fixed and compared with the default `irb` implementation.